### PR TITLE
[kodi] Switched to shared instance of WebSocketClient

### PR DESCRIPTION
**Pull request by cweitkamp**
- Switched to shared instance of WebSocketClient using WebSocketFactory
- Minor code cleanups to fix sat findings

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by cweitkamp** [comment]
@kaikreuzer May I ask you to habe a look if this implementation is okay? Thanks.

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java [hunk 1]
```diff
@@ -74,9 +74,13 @@ public synchronized void open() throws Exception {
         if (isConnected()) {
             logger.warn("connect: connection is already open");
         }
-        if (!client.isStarted()) {
-            client.start();
-        }
+        // if (!client.isStarted()) {
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java]: Not need, right?

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java [hunk 2]
```diff
@@ -96,18 +100,17 @@ public void close() {
             }
             session = null;
         }
-        try {
-            client.stop();
-        } catch (Exception e) {
-            logger.debug("Exception during closing the websocket: {}", e.getMessage(), e);
-        }
+        // try {
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java]: Not need, right?

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java (hunk 1 above)
Comment (martinvw) [reply addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java]: Indeed, you can trust the framework to start the common client for you

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java (hunk 2 above)
Comment (martinvw) [reply addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiClientSocket.java]: 👍

**Comment by cweitkamp** [comment]
> Did you also test it

Yes, of course. But using a single Kodi instance. Will do a second test with two one more different connections.

Unfortunately not today. Maybe tomorrow.

**Comment by martinvw** [comment]
> Unfortunately not today. Maybe tomorrow.

I trust it enough looking at the diff, so when the comments are gone feel free to merge 👍

**Comment by cweitkamp** [comment]
@martinvw Thanks for your trust and your review. I removed the comments and changed the exception type for one method to a more specific one. Merge is up to you.

**Comment by martinvw** [comment]
Thanks for both your PR's

**Pull request merged**